main.tf

```diff
@@ -8,23 +8,31 @@ terraform {
   }
 }
 
+locals {
+
+  # if `worklytics_tenant_id` is null, then empty role assumption policy has effect of allowing
+  # NO ONE to assume the role
+  role_assumption_statements = var.worklytics_tenant_id == null ? [] : {
```

Comment:
terraform fails on this, bc types aren't equivalent between `true` and `false` cases ...

Reply:
and also, turns out that AWS doesn't allow assumption policies that are empty